extensions/grpc/deployment/src/main/java/io/quarkus/grpc/deployment/GrpcClientProcessor.java

```diff
@@ -104,62 +111,93 @@ void discoverInjectedGrpcServices(BeanDiscoveryFinishedBuildItem beanDiscovery,
                         "Invalid @GrpcClient `" + injectionPoint.getTargetInfo() + "` - service name cannot be empty");
             }
 
-            GrpcServiceBuildItem item;
+            GrpcClientBuildItem item;
             if (items.containsKey(serviceName)) {
                 item = items.get(serviceName);
             } else {
-                item = new GrpcServiceBuildItem(serviceName);
+                item = new GrpcClientBuildItem(serviceName);
                 items.put(serviceName, item);
             }
 
             Type injectionType = injectionPoint.getRequiredType();
-            ClassType type;
-            if (injectionType.kind() == Type.Kind.PARAMETERIZED_TYPE) {
-                // Instance<X>
-                type = injectionType.asParameterizedType().arguments().get(0).asClassType();
-            } else {
-                // X directly
-                type = injectionType.asClassType();
+
+            // Programmatic lookup - take the param type
+            if (DotNames.INSTANCE.equals(injectionType.name()) || DotNames.INJECTABLE_INSTANCE.equals(injectionType.name())) {
+                injectionType = injectionType.asParameterizedType().arguments().get(0);
```

Comment:
this will throw some nasty exception if someone tried to use Instance without a type, right?

Reply:
it would be nice to have info where the wrong injection type was met

Reply:
In fact, _"If an injection point of raw type Instance is defined, the container automatically detects the problem and treats it as a definition error."_ (5.6.1. The Instance interface) so ArC should detect this problem just before gRPC extension. I'm not sure it does. I'll check and send a separate PR if needed...

Reply:
See https://github.com/quarkusio/quarkus/pull/17109